cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java

```diff
@@ -486,6 +497,12 @@ private static File updateHiveConfFile(File confFile, File tempDir) {
     conf.setBoolean(Job.MAPREDUCE_JOB_USER_CLASSPATH_FIRST, false);
     conf.setBoolean(MRJobConfig.MAPREDUCE_JOB_CLASSLOADER, false);
 
+    String sparkHome = System.getenv(Constants.SPARK_HOME);
+    if (sparkHome != null) {
+      LOG.debug("Setting spark.home in hive conf to {}", sparkHome);
+      conf.set("spark.home", sparkHome);
```

Comment:
Sounds good, especially spark deployment model only requires spark installed on the edge node only (like cdap-master). Since we already shipped the `spark-assembly.jar` to the explore container, we can have the `{spark.home}/bin/spark-submit` to run our own script, which invoke the `SparkSubmit.main` loaded from the local spark-assembly.jar.